### PR TITLE
Fix for issue #15498

### PR DESCRIPTION
**Pull request by wilsonge**
Pull Request for Issue #15498 .

### Summary of Changes
It's bad practice to typehint against a stdClass instance. Instead it's better to check we don't get `null` back from `JMenu::getActive` or `JMenu::getItem`

### Testing Instructions
Install staging with sample testing data. Go to the featured articles page. Before patch the home page has a url like `JROOT/index.php/19-sample-data-articles/joomla/24-joomla`. After the patch it should be JROOT/index.php

### Documentation Changes Required
None

### CODE REVIEW

File: components/com_content/helpers/legacyrouter.php [hunk 1]
```diff
@@ -91,7 +91,7 @@ public function build(&$query, &$segments)
 		}
 
 		// Are we dealing with an article or category that is attached to a menu item?
-		if (($menuItem instanceof stdClass)
+		if ($menuItem !== null
```
Comment (csthomas) [inline components/com_content/helpers/legacyrouter.php]: Can you add:
```&& isset($query['view'])``` below changed line?

Comment (wilsonge) [inline components/com_content/helpers/legacyrouter.php]: No - that's not related to this bug - it needs to go in a separate Pull Request

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 87ba5af9aff189acfb0f40edcd95e53afafb9c2c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15499">issues.joomla.org/tracker/joomla-cms/15499</a>.</sub>

**Comment by Kubik-Rubik** [comment]
I have tested this item :white_check_mark: successfully on 87ba5af9aff189acfb0f40edcd95e53afafb9c2c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15499">issues.joomla.org/tracker/joomla-cms/15499</a>.</sub>

**Comment by dgt41** [comment]
I have tested this item :white_check_mark: successfully on 87ba5af9aff189acfb0f40edcd95e53afafb9c2c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15499">issues.joomla.org/tracker/joomla-cms/15499</a>.</sub>

**Comment by wilsonge** [comment]
RTC.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15499">issues.joomla.org/tracker/joomla-cms/15499</a>.</sub>

**Comment by dgt41** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15499">issues.joomla.org/tracker/joomla-cms/15499</a>.</sub>
